### PR TITLE
Add convenience method to get targets with a given name

### PR DESCRIPTION
**Pull request by pepibumur**
### Short description 📝
Add convenience method to get targets with a given name

### Solution 📦
Extend `PBXProj.Objects` and add the convenience method in there.

### Implementation 👩‍💻👨‍💻
- [x] Add and test the method
- [x] Move internal helpers from `PBXProj` to `PBXProj.Objects`.

### GIF
gif

---
This change is [Reviewable](https://reviewable.io/reviews/xcodeswift/xcproj/184)

### CODE REVIEW

File: Package.resolved [hunk 1]
```diff
@@ -10,6 +10,15 @@
           "version": "4.1.0"
         }
       },
+      {
+        "package": "BrightFutures",
```
Comment (toshi0383) [inline Package.resolved]: Why are these diffs in this PR?

Comment (allu22) [reply Package.resolved]: +1, it seems these are unnecessary packages.

Comment (pepibumur) [reply Package.resolved]: Ouch, true. I'll revert them.

Comment (toshi0383) [reply Package.resolved]: You reverted it, but restored again.🤣

**Pull request merged**